rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -668,36 +681,29 @@ impl Controller {
         Ok(())
     }
 
-    /// Returns a new system tray menu
-    fn build_system_tray_menu(&self) -> tauri::SystemTrayMenu {
-        // TODO: Refactor this and the auth module so that "Are we logged in"
+    /// Builds a new system tray menu and applies it to the app
+    fn refresh_system_tray_menu(&mut self) -> Result<()> {
+        // TODO: Refactor `Controller` and the auth module so that "Are we logged in?"
         // doesn't require such complicated control flow to answer.
-        // TODO: Show some "Waiting for portal..." state if we got the deep link but
-        // haven't got `on_tunnel_ready` yet.
-        if let Some(auth_session) = self.auth.session() {
+        let menu = if let Some(auth_session) = self.auth.session() {
```

Comment:
We can reduce indentation a bit by using a `let Some(auth_session) = self.auth.session() else { ... }` here

Reply:
I'm having trouble seeing it because there's 3 branches, it returns `menu` and then calls `self.tray.update`